src/js/angular/ttyg/directives/chat-panel.directive.js

```diff
@@ -64,10 +64,21 @@ function ChatPanelComponent(toastr, $translate, TTYGContextService, TTYGService)
                 if (!$scope.chatItem.chatId) {
                     createNewChat();
                 } else {
-                    askQuestion();
+                    askQuestion($scope.chatItem);
                 }
             };
 
+            $scope.regenerate = (chatItem) => {
+                const question = getEmptyChatItem();
+                question.question.message = chatItem.question.message;
```

Comment:
I don't like this. You are revealing the model too much.

Reply:
Added methods to the model that fetch and update its message.